src/plugins/intel_gpu/src/graph/graph_optimizer/reorder_inputs.cpp

```diff
@@ -513,6 +513,23 @@ void reorder_inputs::run(program& p, layout_optimizer& lo, reorder_factory& rf)
         n->recalc_output_layout(true);
     }
 
+    const auto reorder_input_detection_output = [&p, &rf](typed_program_node<detection_output>& detection_output_node) {
+        if (detection_output_node.get_preferred_impl_type() == impl_types::cpu) {
+            auto detection_output_prim = detection_output_node.get_primitive();
+
+            for (size_t i = 0; i < detection_output_node.get_dependencies().size(); i++) {
+                auto& input = detection_output_node.get_dependency(i);
+                auto new_input = rf.get_reorder(input.id(),
+                                                input.get_output_layout(),
+                                                layout{ data_types::f32, format::bfyx, input.get_output_layout().get_tensor() });
+
+                if (new_input.first) {
+                    p.add_intermediate(new_input.first, detection_output_node, i, !new_input.second);
+                }
+            }
+        }
+    };
+
```

Comment:
What about let reorder be attached and removing the reorder from remove_redundant_reorder?

Reply:
Yes, it's alternative way, but in order to do, we need to add new logic which handling cross data type case in remove_reduntant_reorder. I think it's already complicated and after adding new logic, we need full test for validation too. It seems the burden is more than we can get.